ITEvents/Screens/Search tab/SearchPresenter.swift

```diff
@@ -19,39 +23,70 @@ class SearchPresenter: ISearchPresenter {
         self.selectedEventService = selectedEventService
         self.userSettingsService = userSettingsService
         self.dateFormatterService = dateFormatterService
+        
+        //TODO implement request cancelation
+        searchEventsDebounced = debounce(
+            delay: DispatchTimeInterval.seconds(2),
+            queue: DispatchQueue.main,
+            action: searchEventDebouncedAction
+        )
     }
     
     func setup() {
-        activate()
+        searchEventDebouncedAction()
+    }
+    
+    func selectEvent(with eventId: Int) {
+        selectedEventService.selectedEvent = events.first(where: { $0.id == eventId })
+    }
+
+    func searchMoreEvents() {
+        if eventsTotal == events.count {
+            return
+        }
+        
+        view.toggleProgressIndicator(shown: true)
         
-        eventDataService.fetchEvents(then: { fetchedEvents in
-            self.events = fetchedEvents
-            let eventCollectionCellViewModels = self.events.map { self.createEventCollectionCellViewModel(event: $0)}
-            self.view.setEvents(eventCollectionCellViewModels)
+        eventDataService.searchEvents(indexRange: events.count..<events.count + 10, parameters: searchParameters, then: { fetchedEvents, total in
```

Comment:
we can create a function ```onEventSearched(fetchedEvents, total)``` to get rid of copy/pasted code in callbacks.
Then we can use it the following way here and in ```searchEventDebouncedAction``` function:
```swift
eventDataService.searchEvents(indexRange: events.count..<events.count + 10, 
                              parameters: searchParameters, 
                              then: onEventSearched)
```